client/src/main/scala/io/delta/sharing/client/DeltaSharingProfileProvider.scala

```diff
@@ -17,22 +17,136 @@
+        "Cannot find the 'clientSecret' field in the profile file")
+    }
+  }
+}
+
+case class
+OAuthClientCredentialsDeltaSharingProfile(override
+                                          val shareCredentialsVersion: Option[Int],
+                                          override
+                                          val endpoint: String = null,
+                                          tokenEndpoint: String = null,
+                                          clientId: String = null,
+                                          clientSecret: String = null,
+                                          scope: Option[String] = None
+                                         ) extends DeltaSharingProfile
+
+
+private[client]
+class DeltaSharingProfileDeserializer extends JsonDeserializer[DeltaSharingProfile] {
+  override def deserialize(p: JsonParser, ctxt: DeserializationContext): DeltaSharingProfile = {
+    val node: ObjectNode = p.getCodec.readTree(p).asInstanceOf[ObjectNode]
+
+    if (node.has("type") &&
+      node.get("type").asText().toLowerCase(Locale.ROOT) == "oauth_client_credentials") {
+      // Deserialize as OAuthClientCredentialsDeltaSharingProfile
+      OAuthClientCredentialsDeltaSharingProfile(
+        shareCredentialsVersion = Option(node.get("shareCredentialsVersion")).map(_.asInt()),
+        endpoint = Option(node.get("endpoint")).map(_.asText()).orNull,
+        tokenEndpoint = Option(node.get("tokenEndpoint")).map(_.asText()).orNull,
+        clientId = Option(node.get("clientId")).map(_.asText()).orNull,
+        clientSecret = Option(node.get("clientSecret")).map(_.asText()).orNull,
+        scope = Option(node.get("scope")).map(_.asText())
+      )
+
+    } else {
+      // Deserialize as BearerTokenDeltaSharingProfile
+      BearerTokenDeltaSharingProfile(
+        shareCredentialsVersion = Option(node.get("shareCredentialsVersion")).map(_.asInt()),
+        endpoint = Option(node.get("endpoint")).map(_.asText()).orNull,
+        bearerToken = Option(node.get("bearerToken")).map(_.asText()).orNull,
+        expirationTime = Option(node.get("expirationTime")).map(_.asText()).orNull
+      )
+    }
+  }
+}
+
 
 object DeltaSharingProfile {
-  val CURRENT = 1
+  val CURRENT = 2
+
+  // constructor to ensure backward compatibility for DeltaSharingProfile instantiation
+  def apply(shareCredentialsVersion: Some[Int],
+            endpoint: String,
+            bearerToken: String,
+            expirationTime: String = null): DeltaSharingProfile = {
+    new BearerTokenDeltaSharingProfile(shareCredentialsVersion,
+      endpoint, bearerToken, expirationTime)
+  }
+
+  private[client] def validate(profile: DeltaSharingProfile): Unit = {
+    if (profile.shareCredentialsVersion.isEmpty) {
+      throw new IllegalArgumentException(
+        "Cannot find the 'shareCredentialsVersion' field in the profile file")
+    }
+
+    if (profile.shareCredentialsVersion.get > DeltaSharingProfile.CURRENT) {
```

Comment:
do we need to check that the version is 1 for `BearerTokenDeltaSharingProfile`?

Reply:
Thanks for the comment. addressed.